src/core/expression/qgsexpressionutils.h

```diff
@@ -291,6 +292,18 @@ class QgsExpressionUtils
       return QgsInterval();
     }
 
+    static QgsGradientColorRamp getRamp( const QVariant &value, QgsExpression *parent, bool report_error = false )
+    {
+      if ( value.canConvert<QgsGradientColorRamp>() )
+        return value.value<QgsGradientColorRamp>();
+
+      // If we get here then we can't convert so we just error and return invalid.
+      if ( report_error )
+        parent->setEvalErrorString( QObject::tr( "Cannot convert '%1' to QgsGradientColorRamp" ).arg( value.toString() ) );
```

Comment:
@DelazJ , trying to stick to what was established elsewhere (see getGeometry(), getFeature()).